client/my-sites/plans/current-plan/index.jsx

```diff
@@ -113,6 +124,13 @@ class CurrentPlan extends Component {
 				<QuerySites siteId={ selectedSiteId } />
 				<QuerySitePlans siteId={ selectedSiteId } />
 				{ shouldQuerySiteDomains && <QuerySiteDomains siteId={ selectedSiteId } /> }
+				{ doJetpackPlanSetup && (
+					<JetpackSetupRunner
+						key={ /* Force remount on site change */ selectedSiteId }
```

Comment:
Hmm, shouldn't that be handled by a lifecycle method in `JetpackSetupRunner` that watches out for changes of `siteId`?

Reply:
It's pretty difficult to properly reinitialize components that aren't remounted. This is a way of ensuring that, if the siteId were changed, the component would be thrown out and mounted fresh– hat tip @jsnajdr p4TIVU-910-p2